src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounter.cs

```diff
@@ -300,7 +300,7 @@ public string MachineName
             set
             {
                 if (!SyntaxCheck.CheckMachineName(value))
-                    throw new ArgumentException(SR.Format(SR.InvalidParameter, "machineName", value));
+                    throw new ArgumentException(SR.Format(SR.InvalidParameter, nameof(MachineName), value), nameof(MachineName));
```

Comment:
For the argument names, do we want to be using nameof(PropertyName) or nameof(value)?

Reply:
I prefer `nameof(value)` personally, but don't have a strong preference either way as long as it's consistent throughout.

I see that `CategoryName` is using `throw new ArgumentNullException(nameof(value))` while using the name of the property in the exception message:

https://github.com/dotnet/corefx/blob/53640cd9640d4e7c1f41e26b6e2ddb4feb9e837b/src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterCategory.cs#L75-L79

I suppose we should do the same here.

As an aside, this should probably be using `SR.InvalidProperty` not `SR.InvalidParameter`.

So I'll change it to: 

```c#
throw new ArgumentException(SR.Format(SR.InvalidProperty, nameof(MachineName), value), nameof(value));
```

Reply:
I have asked folks to use `nameof(PropertyName)` since from the caller's point of view, they don't see `value`. Also, in an exception message, if there's no line number for some reason it may help locate where the exception was from. Is there a reason why `value` might be better?

Reply:
> I have asked folks to use nameof(PropertyName) since from the caller's point of view

The caller doesn't see method parameter names, either (unless explicitly specified with optional name syntax).  And the name of the set argument actually is "value", so it doesn't seem different to me than using "value" for a method like BitConverter.GetBytes.

> Is there a reason why value might be better?

Because it actually is the argument name? :smile:

@KrzysztofCwalina, @terrajobst, do we have guidance on this?

Reply:
Sometimes exception messages leak to users, and I could imagine that "Argument 'UserName' is invalid." might be more useful to them than "Argument 'value' is invalid".

I don't feel strongly and I'm happy to go with the consensus as long as we try to be consistent.

Reply:
> I don't feel strongly and I'm happy to go with the consensus as long as we try to be consistent.

Ditto